### PR TITLE
Progress params

### PR DESCRIPTION
**Pull request by algesten**
This PR is in flux and shouldn't land before the proposed changes to LS spec.

https://github.com/Microsoft/vscode-languageserver-node/pull/261/files

### CODE REVIEW

**Comment by algesten** [comment]
Seems RLS is using an ancient 0.16.0 version of languageserver-types :/

**Comment by Marwes** [comment]
0.16 is not that old (it's from November) and all the changes in between has just been small bug fixes or some small extensions to the API so I wouldn't worry that much about it :).  Feel free to reopen if you like (or wait until it gets merged to the spec).

File: src/lib.rs [hunk 1]
```diff
@@ -2375,6 +2375,32 @@ pub struct MarkupContent {
...
+ * to indicate progress.
+ */
+#[derive(Debug, PartialEq, Deserialize, Serialize)]
+pub struct ProgressParams {
+    // A unique identifier to associate multiple progress notifications with the same progress.
+    pub id: String,
+
+    // The title of the progress.
+    // This should be the same for all ProgressParams with the same id.
+    pub title: String,
```
Comment (Marwes) [inline src/lib.rs]: SHould be optional I believe

File: src/lib.rs [hunk 2]
```diff
@@ -2375,6 +2375,32 @@ pub struct MarkupContent {
...
+    // If unset, the previous progress message (if any) is still valid.
+    pub message: Option<String>,
+
+    // Optional progress percentage to display.
+    // If unset, the previous progress percentage (if any) is still valid.
+    pub percentage: Option<f64>,
+
+    // Set to true on the final progress update.
+    // No more progress notifications with the same ID should be sent.
+    done: bool,
```
Comment (Marwes) [inline src/lib.rs]: Optional

**Comment by algesten** [comment]
cool. i'll open a new PR against master while i keep my 0.16 version locally.

**Pull request closed without merging**